### ISSUE TITLE
If file is corrupt, make BF program skip it

### ISSUE THREAD

**Issue by LandonTClipp** [TerraFusion/basicFusion#165]
I recently talked to Dr. Di Girolamo about this. He said that if the BF program encounters a corrupt file (basically, it fails to open it), it should skip that file and move on, marking the event in some kind of error output. If it can be avoided, errors like this should not be fatal.

**Comment by kyang2014** [comment]
This definitely needs to be done before our final processing but not now. We need to check and revise our code carefully to make sure the skipping file will not cause  bad behavior downstream. Ideally, the corrupted files should not be in the input granules!!

**Comment by LandonTClipp** [comment]
Dr. Zhao has asked me to implement this behavior. I will do this now.

**Comment by kyang2014** [comment]
Make sure to test with valgrind if the skipping causes any memory leaking.

**Comment by kyang2014** [comment]
Using valgrind to check the memory leaking should be the general practice for any non-trivial change of the code.  You can do one checking for several changes if you are confident about your changes.

**Comment by LandonTClipp** [comment]
Good idea, I will perform this check now.

**Comment by LandonTClipp** [comment]
Shared HDF4 library files not available on BW at the moment. Valgrind needs an executable linked with shared libraries to track mallocs (a statically-compiled executable will result in an erroneous "0 leaks possible" message from valgrind). Pending BW staff email.

**Comment by kyang2014** [comment]
Why not using roger to run valgrind?  you just need to check the orbits that have the corrupt cases. Roger should be ideal for this case.

**Comment by LandonTClipp** [comment]
I've been hopping back and forth between development on ROGER and BW. Every time I do that it seems there is some issue with porting the program back and forth that needs to be resolved. Remaining on BW for the moment because I've been doing testing there. Plus, valgrind results on ROGER don't necessarily equal valgrind results on BW.

**Comment by kyang2014** [comment]
Running valgrind at BW frequently  may affect other users. The memory leaking check we require at the two machines should be the same.

**Comment by LandonTClipp** [comment]
Valgrind is easy to run on BW (I request compute nodes, so will not affect other users). These 1-node 1-processor jobs backfill very quickly. I will just go ahead and check on ROGER since this issue with HDF libraries on BW may not be resolved immediately.

**Comment by LandonTClipp** [comment]
Valgrind reports no definite losses. A 1 byte invalid read was discovered and fixed.
